pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/segment/SegmentSizeBasedFlushThresholdUpdater.java

```diff
@@ -107,11 +113,16 @@ public void updateFlushThreshold(@Nonnull LLCRealtimeSegmentZKMetadata newSegmen
     }
 
     if (numRowsConsumed < numRowsThreshold) {
-      LOGGER.info("Segment {} reached time threshold, setting thresholds for segment {} from previous segment",
-          committingSegmentZKMetadata.getSegmentName(), newSegmentZKMetadata.getSegmentName());
       // TODO: add feature to adjust time threshold as well
-      // TODO: setting sizeThresholdToFlushSegment instead of numRowsConsumed, otherwise we can end up oscillating back and forth
-      newSegmentZKMetadata.setSizeThresholdToFlushSegment(committingSegmentZKMetadata.getSizeThresholdToFlushSegment());
+      // If we set new threshold to be numRowsConsumed, we might keep oscillating back and forth between doubling limit and time threshold being hit
+      // If we set new threshold to be committingSegmentZKMetadata.getSizeThresholdToFlushSegment(), we might end up using a lot more memory than required for the segment
+      // Using a minor bump strategy, until we add feature to adjust time
+      // We will only slightly bump the threshold based on numRowsConsumed
+      long targetSegmentNumRows = (long) (numRowsConsumed * 1.1);
```

Comment:
Please make a static int for 1.1 and use it for the arithmetic as well as the log message